packages/solana/lib/src/subscription_client/subscription_client.dart

```diff
@@ -17,155 +15,35 @@ import 'package:solana/src/subscription_client/notification_message.dart';
-    final id = message.subscription;
-    final subscriptionManager = _attachedSubscriptionManagers[id];
-    if (subscriptionManager == null) {
-      throw BadStateException(
-        'received a message for subscription $id, but it was not found',
-      );
-    } else {
-      subscriptionManager.add(message.value);
-    }
-  }
-
-  void _attachSubscriptionManager(int requestId, int subscriptionId) {
-    final subscriptionManager = _unattachedSubscriptionManagers.remove(
-      requestId,
-    );
-    if (subscriptionManager == null) {
-      throw const BadStateException('cannot attach subscription manager');
-    } else {
-      _attachedSubscriptionManagers[subscriptionId] = subscriptionManager;
-      subscriptionManager.subscriptionId = subscriptionId;
-    }
-  }
-
-  void _sendRequest(int id, String method, List<dynamic>? params) {
-    final payload = json.encode(<String, dynamic>{
-      'jsonrpc': '2.0',
-      'id': id,
-      'method': method,
-      if (params != null) 'params': params,
-    });
-
-    _webSocket.add(payload);
-  }
-
-  void _createSubscription(int id, String method, List<dynamic>? params) {
-    _sendRequest(id, '${method}Subscribe', params);
-  }
-
-  void _cancelSubscription(String method, int id) {
-    final subscriptionManager = _attachedSubscriptionManagers.remove(id);
-    // This would most certainly mean that something went wrong or unexpectedly
-    // we throwing an exception seems appropriate
-    if (subscriptionManager == null) {
-      throw const BadStateException(
-        'tried to cancel a non existing subscription',
-      );
-    }
-    // Single shot subscriptions are cancelled without sending the
-    // unsubscribe request. So we check that this is not a single shot
-    // subscription before sending the unsubscribe request.
-    if (subscriptionManager.isSingleShot == false) {
-      _sendRequest(_lastRequestId++, '${method}Unsubscribe', <int>[id]);
-    }
-  }
-
-  Stream<T> _subscribe<T>(
-    String method, {
-    List<dynamic>? params,
-    bool singleShot = false,
-  }) {
-    late final SubscriptionManager<T> subscriptionManager;
-    final requestId = _lastRequestId++;
-
-    subscriptionManager = SubscriptionManager<T>(
-      onListen: () {
-        _createSubscription(requestId, method, params);
-      },
-      onCancel: (int subscriptionId) {
-        _cancelSubscription(method, subscriptionId);
-      },
-      singleShot: singleShot,
-    );
-
-    _unattachedSubscriptionManagers[requestId] = subscriptionManager;
-
-    return subscriptionManager.stream;
-  }
-
-  /// Connect to the websocket at [url] node.
-  static Future<SubscriptionClient> connect(String url) async =>
-      SubscriptionClient._(await WebSocket.connect(url));
-
-  /// Dispose this object and cancel any existing subscription.
-  void dispose() {
-    _subscription.cancel();
-  }
+  /// Subscribe to all incoming messages.
+  Stream<SubscriptionMessage> allMessages() => _stream.map(_parse);
```

Comment:
I really don't see the use case. Normally the instantiator of this class will not care about handling messages manually. So this is probably not needed.

Reply:
There is a use case in the cryptoplease app. We need to have a single entrypoint, as the control over the connection should still be done by the client.

In case of using multiple subscriptions, we can simply do something like:

```dart
client.allMessages().listen(null, onDone: () {
  // connection is closed, reconnect
});
```

Also, it can be useful for centralized logging / error handling etc.

Reply:
Yes, I like it. I was also wondering how we would do this. But this is precisely the way. Thanks.